w09h03/test/pgdp/filter/DGTest.java

```diff
@@ -0,0 +1,5539 @@
+        VideoContainer in = new VideoContainer(fp);
+        VideoContainer in1 = new VideoContainer(fp1);
+
+        Object[] array = in1.frameStream.limit(length).toArray();
+
+        // limitiere Laufzeit
+        in.limit(length);
+
+        Object[] output = in.frameStream.toArray();
+
+        Assertions.assertEquals(output.length, length);
+
+        for(int i = 0; i < length; i++){
+            for(int j = 0; j < fp.getWidth(); j++){
+                for(int k = 0; k < fp.getHeight(); k++){
+                    Assertions.assertEquals(((Frame) array[i]).getPixels().getRGB(j,k), ((Frame) output[i]).getPixels().getRGB(j,k));
+                }
+            }
+        }
+    }
+
+    @Test
+    @DisplayName("Test DG 123 VideoContainer Limit noot auf 1 Frames")
+    public void test123() throws FileNotFoundException, IllegalVideoFormatException {
+
+        int length = 1;
+
+        FrameProvider fp = new FrameProvider("noot.mp4");
+        FrameProvider fp1 = new FrameProvider("noot.mp4");
+        VideoContainer in = new VideoContainer(fp);
+        VideoContainer in1 = new VideoContainer(fp1);
+
+        Object[] array = in1.frameStream.limit(length).toArray();
+
+        // limitiere Laufzeit
+        in.limit(length);
+
+        Object[] output = in.frameStream.toArray();
+
+        Assertions.assertEquals(output.length, length);
+
+        for(int i = 0; i < length; i++){
+            for(int j = 0; j < fp.getWidth(); j++){
+                for(int k = 0; k < fp.getHeight(); k++){
+                    Assertions.assertEquals(((Frame) array[i]).getPixels().getRGB(j,k), ((Frame) output[i]).getPixels().getRGB(j,k));
+                }
+            }
+        }
+    }
+
+    @Test
+    @DisplayName("Test DG 124 VideoContainer Limit noot auf 0 Frames")
+    public void test124() throws FileNotFoundException, IllegalVideoFormatException {
+
+        int length = 0;
+
+        FrameProvider fp = new FrameProvider("noot.mp4");
+        FrameProvider fp1 = new FrameProvider("noot.mp4");
+        VideoContainer in = new VideoContainer(fp);
+        VideoContainer in1 = new VideoContainer(fp1);
+
+        // limitiere Laufzeit
+
+        try{
+            in.limit(length);
+        }
+        catch(Exception e){
+            Assertions.fail();
+        }
+
+
+        Object[] output = in.frameStream.toArray();
+
+        Assertions.assertEquals(output.length, length);
+    }
+
+    @Test
+    @DisplayName("Test DG 125 VideoContainer Limit noot auf -1 Frames")
+    public void test125() throws FileNotFoundException, IllegalVideoFormatException {
+
+        int length = -1;
+
+        FrameProvider fp = new FrameProvider("noot.mp4");
+        FrameProvider fp1 = new FrameProvider("noot.mp4");
+        VideoContainer in = new VideoContainer(fp);
+        VideoContainer in1 = new VideoContainer(fp1);
+
+        // limitiere Laufzeit
+        try{
+            in.limit(length);
+        }
+        catch(Exception e){
+            Assertions.fail();
```

Comment:
Why should limit throw an exception?

Reply:
We should never throw exceptions, except they are required. That was the clear instruction, thus I think this is acceptable.

Reply:
Ist das nicht bezogen auf den Iterator?

Reply:
> We should never throw exceptions, except they are required. That was the clear instruction, thus I think this is acceptable.

I thought that the limitation is restricted to the iterator.

Reply:
These are general instructions which had been like for every last task. Not task-explicit instructions. I did not read any explicit changes to this iterator.

Reply:
I always assert the parameters and all tests have passed so far. I'll ask on zulip then.